core/src/SYCL/Kokkos_SYCL_Instance.hpp

```diff
@@ -228,6 +232,12 @@ class SYCLInternal {
   static void fence_helper(WAT& wat, const std::string& name,
                            uint32_t instance_id);
 
+  const static size_t m_usm_pool_size =
+      30;  // TODO joe: arbitrary, should be configurable?
```

Comment:
Let's see in a test similar to https://github.com/kokkos/kokkos/issues/3670 what the maximum number of kernels is that we can schedule concurrently and choose the number with respect to that.

Reply:
We should still run some experiments to see what a good number is.